File: frontend/src/microservices/feedback/objects/FeedbackThreadListPlannerResponse.ts
// 本文件定义 feedback 模块的 `FeedbackThreadListPlannerResponse`，作为线程列表响应数据并提供 JSON 编解码。

import type { FeedbackThreadDetailsPlannerResponse } from './FeedbackThreadDetailsPlannerResponse'

export type FeedbackThreadListPlannerResponse = {
  threads: FeedbackThreadDetailsPlannerResponse[]
}
export const feedbackThreadListPlannerResponseFromJson = (json: string): FeedbackThreadListPlannerResponse =>
  JSON.parse(json) as FeedbackThreadListPlannerResponse

export const feedbackThreadListPlannerResponseToJson = (value: FeedbackThreadListPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/feedback/objects/FeedbackThreadListPlannerResponse.scala
// FeedbackThreadListPlannerResponse：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FeedbackThreadListPlannerResponse(threads: List[FeedbackThreadDetailsPlannerResponse])
object FeedbackThreadListPlannerResponse:
  import ContentSourceJsonCodecs.given
  given sourceEncoder: Encoder[FeedbackThreadListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FeedbackThreadListPlannerResponse] = deriveDecoder
